### PR TITLE
[Wiki] Updated 'Module Library' section to latest

### PR DESCRIPTION
**Pull request by AlexanderSehr**
# Description

- Updated 'Module Library' section to latest

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Update to documentation

### CODE REVIEW

**Comment by ahmadabdalla** [comment]
@AlexanderSehr are there PRs you'd like to be merged first before these doc ones get updated?

**Comment by AlexanderSehr** [comment]
> @AlexanderSehr are there PRs you'd like to be merged first before these doc ones get updated?

Hey @ahmadabdalla, there aren't. We're trying to push this bad boy through asap. In theory we could wait for the singular change, but personally I think it doesn't make much of a difference.
cc: @eriqua

**Comment by eriqua** [comment]
> > @AlexanderSehr are there PRs you'd like to be merged first before these doc ones get updated?
> 
> Hey @ahmadabdalla, there aren't. We're trying to push this bad boy through asap. In theory we could wait for the singular change, but personally I think it doesn't make much of a difference. cc: @eriqua

Should we merge these to dev or to main?

**Comment by AlexanderSehr** [comment]
> > > @AlexanderSehr are there PRs you'd like to be merged first before these doc ones get updated?
> > 
> > 
> > Hey @ahmadabdalla, there aren't. We're trying to push this bad boy through asap. In theory we could wait for the singular change, but personally I think it doesn't make much of a difference. cc: @eriqua
> 
> Should we merge these to dev or to main?

That's my point. I think main is OK, as these are just docs updates that are not dependent on anything - as opposed to the module changes that all need to be done / merged in one go

File: docs/wiki/The library - Module design.md [hunk 1]
```diff
@@ -48,10 +48,12 @@ They can be deployed in different configurations just by changing the input para
...
   1. **Diagnostic logs** and **metrics** (you can have them sent to one ore more of the following destination types: storage account, log analytics and event hub).
-  2. Resource and child resource level **RBAC** (for example, providing data contributor access on a storage account; granting file share/blob container level access in a storage account)
-  3. **Tags** (as objects)
-  4. **Locks**
-  5. **Private Endpoints** (if supported)
+  1. Resource and child-resource level **RBAC** (for example, providing data contributor access on a storage account; granting file share/blob container level access in a storage account)
+  1. **Tags** (as objects)
+  1. **Locks**
+  1. **Private Endpoints** (if supported)
+  1. **User-defined Types** (if supported)
```
Comment (eriqua) [inline docs/wiki/The library - Module design.md:55]: 🪄

File: docs/wiki/The library - Module design.md [hunk 2]
```diff
@@ -107,12 +103,13 @@ module server_databases 'databases/main.bicep' = [for (database, index) in datab
 
 Use the following naming standard for module files and folders:
 
-- Module folders are in camelCase and their name reflects the main resource type of the Bicep module they are hosting (e.g., `storageAccounts`, `virtualMachines`).
+- A module's 'Provider namespace' folder is lowercase, avoids the `Microsoft.` prefix and uses a `-` as a separator whenever the API reference would have an upper case (for example `'Microsoft.RecoveryServices'` would have a folder name `'recovery-service'`)
```
Comment (eriqua) [inline docs/wiki/The library - Module design.md:106]:
```suggestion
- A module's 'Provider namespace' folder is lowercase, avoids the `Microsoft.` prefix and uses a `-` as a separator whenever the API reference would have an upper case (for example `'Microsoft.RecoveryServices'` would have a folder name `'recovery-services'`)
```

File: docs/wiki/The library - Module design.md (hunk 1 above)
Comment (AlexanderSehr) [reply docs/wiki/The library - Module design.md:55]: I'm confused, is there a todo? :D 🍡

Comment (eriqua) [reply docs/wiki/The library - Module design.md:55]: Did you mean user-defined types or something else? I felt like they were not intended to go in this list. In general I'd wait for them to be introduced before documenting them as part of the intended design.

File: docs/wiki/The library - Module design.md [hunk 3]
```diff
@@ -215,11 +216,41 @@ The `builtInRoleNames` variable contains the list of applicable roles for the sp
...
+@sys.description('Optional. The conditions on the role assignment. This limits the resources it can be assigned to. e.g.: @Resource[Microsoft.Storage/storageAccounts/blobServices/containers:ContainerName] StringEqualsIgnoreCase "foo_storage_container".')
+param condition string = ''
+
+@sys.description('Optional. Version of the condition.')
+@allowed([
+  '2.0'
+])
+param conditionVersion string = '2.0'
+
+@sys.description('Optional. Id of the delegated managed identity resource.')
```
Comment (eriqua) [inline docs/wiki/The library - Module design.md:251]:
```suggestion
@sys.description('Optional. ID of the delegated managed identity resource.')
```

File: docs/wiki/The library - Module design.md (hunk 1 above)
Comment (AlexanderSehr) [reply docs/wiki/The library - Module design.md:55]: I think this should be Managed Identities or Customer-Managed Keys :D Will doublecheck. Definitely not User-defined types...

Comment (AlexanderSehr) [reply docs/wiki/The library - Module design.md]: @eriqua fixed

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**